src/eventDispatchers/mouseEventHandlers/mouseDown.js

```diff
@@ -29,122 +30,193 @@ export default function (evt) {
 
   let tools;
   const eventData = evt.detail;
-  const element = eventData.element;
-  const coords = eventData.currentPoints.canvas;
-  const distance = 6;
+  const element = evt.detail.element;
+  const coords = evt.detail.currentPoints.canvas;
 
-  // TODO: instead of filtering these for every interaction, we can change our
-  // State's structure to always know these values.
-  // Filter out disabled and enabled
+  // High level filtering
   tools = getInteractiveToolsForElement(element, getters.mouseTools());
-  // Filter out tools that do not match mouseButtonMask
   tools = tools.filter((tool) =>
     isMouseButtonEnabled(eventData.which, tool.options.mouseButtonMask)
   );
-  tools = getToolsWithDataForElement(element, tools);
 
-  // Find tools with handles we can move
-  const toolsWithMoveableHandles = tools.filter((tool) => {
-    const toolState = getToolState(eventData.element, tool.name);
+  // Check if any tool has a special reason to grab the current event
+  const specialTools = tools.filter(
+    (tool) =>
+      typeof tool.mouseDownCallback === 'function' &&
+      typeof tool.isValidTarget === 'function' &&
+      tool.mode === 'active' &&
```

Comment:
I think restricting this to active tools makes sense, more of a question than a comment: is there any tool which requires mouseDown functionality whilst passive and not hovering over a handle? I don't think there is, so feel this is correct.

Reply:
Edit: quick mockup if we went with my first solution:

```
const specialTools = tools.filter(
  (tool) => tool instanceof 'SpecialTool' &&
  tool.isValidTarget(evt)
);
```